util/incoming-connections.js

```diff
@@ -68,35 +67,52 @@ module.exports = function (config) {
     incoming = Object.keys(defaultPorts).map((service) => {
       return {
         service,
-        interfaces: Object.values(interfaces).reduce((acc, val) => {
-          // Future TODO: replace with shiny new `Array.prototype.flat()`.
-          return acc.concat(val)
-        }, []).filter(item => {
-          // We want to avoid scoped IPv6 addresses since they don't seem to
-          // play nicely with the Node.js networking stack. These addresses
-          // often start with `fe80` and throw EINVAL when we try to bind to
-          // them.
-          return item.scopeid == null || item.scopeid === 0
-        }).map(item => {
+        interfaces: Object.values(interfaces)
+          .reduce((acc, val) => {
+            // Future TODO: replace with shiny new `Array.prototype.flat()`.
+            return acc.concat(val)
+          }, [])
+          .filter(item => {
+            // We want to avoid scoped IPv6 addresses since they don't seem to
+            // play nicely with the Node.js networking stack. These addresses
+            // often start with `fe80` and throw EINVAL when we try to bind to
+            // them.
+            return item.scopeid == null || item.scopeid === 0
+          })
+          .reduce((acc, cur) => {
+            // It's possible to have two interfaces with the same IP address,
+            // but we don't want to try to listen on both of them. This only
+            // adds the interface to the list if it hasn't already been added.
+            const found = acc.find((item) =>
+              item.address === cur.address
+            )
+
+            if (found == null) {
+              acc.push(cur)
+            }
+
+            return acc
```

Comment:
To simplify this code, I would probably refactor this arrow function to be a named function `uniqueAddresses` that's defined as a top-level function and then I'd use it as `reduce(uniqueAddresses)`. I would do the same for the other parts of this chain of array methods.